slick-codegen/src/main/scala/slick/codegen/AbstractSourceCodeGenerator.scala

```diff
@@ -61,10 +61,18 @@ abstract class AbstractSourceCodeGenerator(model: m.Model)
       if(hlistEnabled) values.mkString(" :: ") + " :: HNil"
       else if (values.size == 1) values.head
       else if(values.size <= 22) s"""(${values.mkString(", ")})"""
-      else throw new Exception("Cannot generate tuple for > 22 columns, please set hlistEnable=true or override compound.")
+      else values.mkString("(" , " :: ", " :: HNil)")
+      //throw new Exception("Cannot generate tuple for > 22 columns, please set hlistEnable=true or override compound.")
     }
 
-    def factory   = if(columns.size == 1) TableClass.elementType else s"${TableClass.elementType}.tupled"
+    def factory   = if(columns.size == 1) TableClass.elementType 
+                    else if(columns.size > 22 & ! hlistEnabled ){
+                      columns.map(c => 
+                        {
+                          s"x(${columns.indexOf(c)})"
+                      }).mkString(s"({case x => ${TableClass.elementType}(" , "," , ")})" )
```

Comment:
I suppose this should always work, so we don't even need the if and the else branches? also, you don't need `case` and `{ ... }` here, you can just leave it out